File: go/bob/bob.go
// Package bob is a package for responding to Bob
package bob

import (
	"strings"
	"unicode"
)

// Hey returns a canned response from Bob given a type of remark.
func Hey(remark string) string {

	//trim spaces
	trimmed := strings.TrimSpace(remark)

	if len(trimmed) == 0 {
		return "Fine. Be that way!"
	}
	punctuation := trimmed[len(trimmed)-1:]

	switch punctuation {
	case "?":
		if isYelling(remark) {
			return "Calm down, I know what I'm doing!"
		}
		return "Sure."
	default:
		if isYelling(remark) {
			return "Whoa, chill out!"
		}
		return "Whatever."
	}
}

func isYelling(remark string) bool {
	if remark == strings.ToUpper(remark) && containsCharacter(remark) {
		return true
	}
	return false
}

func containsCharacter(remark string) bool {
	for _, v := range remark {
		if unicode.IsLetter(v){
			return true
		}
	}
	return false
}

File: python/bob/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    is_question = hey_bob[-1:] == '?'
    is_yelling = hey_bob.isupper()
    is_silent = not any(c.isalnum() for c in hey_bob)
    if is_silent and not is_question:
        return 'Fine. Be that way!'
    if is_yelling and is_question:
        return "Calm down, I know what I'm doing!"
    if is_question:
        return 'Sure.'
    if is_yelling:
        return 'Whoa, chill out!'

    return 'Whatever.'